.github/workflows/pull-request-with-code.yml

```diff
@@ -17,35 +17,63 @@ on:
       - '**/*.toml'
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

Comment:
New thing, this will cancel ongoing PR builds if someone pushes new commits to it